src/lib/storage/zero_suppression/zs_vector_meta_info.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <cstdint>
+#include <optional>
+
+namespace opossum {
+
+/**
+ * @brief Meta information about an unencoded vector
+ *
+ * Some encoders can utilize additional information
+ * about the vector which is to be encoded.
```

Comment:
Is this final? If not, does it have a maximum length? Can the length of possible subclasses be checked at compile time?

Reply:
This can be extended if new encoding schemes can make use of other information.